src/battle_ai_util.c

```diff
@@ -493,6 +493,19 @@ bool32 IsDamageMoveUnusable(u32 move, u32 battlerAtk, u32 battlerDef)
         if (!gDisableStructs[battlerAtk].isFirstTurn)
             return TRUE;
         break;
+    case EFFECT_FOCUS_PUNCH:
+        if (HasDamagingMove(battlerDef) && !((gBattleMons[battlerAtk].status2 & STATUS2_SUBSTITUTE)
+         || IsBattlerIncapacitated(battlerDef, aiData->abilities[battlerDef])
+         || gBattleMons[battlerDef].status2 & (STATUS2_INFATUATION | STATUS2_CONFUSION)))
+         // TODO: || IsPredictedToSwitch(battlerDef, battlerAtk)
+            return TRUE;
+        // If AI could Sub and doesn't have a Sub, don't Punch yet
+        for (i = 0; i < MAX_MON_MOVES; i++)
+        {
+            if (gMovesInfo[gBattleMons[battlerAtk].moves[i]].effect == EFFECT_SUBSTITUTE && !(gBattleMons[battlerAtk].status2 & STATUS2_SUBSTITUTE))
+                return TRUE;
+        }
```

Comment:
we have a function for that. `HasMoveEffect`

Reply:
Sweet, will refactor! :)

Reply:
Don't forget about this.

Though a comment brought up by Jasper which is very valid is that `CheckBadMove` and `IsDamageMoveUnusable` that would fail on the next turn so including actual AI move logic goes against the idea. I don't think we should change it right now but just something to keep in mind for the future.

Reply:
I actually made a commit for this yesterday, was just waiting to push until we'd resolved the other business :)

I think this is a good segue into future discussion for sure. I'm not sure if a `IsMoveReasonable` function makes sense, or a new AI flag, or something different. Something to think about for sure.